server/catgenome/src/main/java/com/epam/catgenome/dao/lineage/LineageTreeNodeDao.java

```diff
@@ -0,0 +1,154 @@
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.epam.catgenome.dao.lineage;
+
+import com.epam.catgenome.dao.DaoHelper;
+import com.epam.catgenome.entity.lineage.LineageTreeNode;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcDaoSupport;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Getter
+@Setter
+@NoArgsConstructor
+@AllArgsConstructor
+public class LineageTreeNodeDao extends NamedParameterJdbcDaoSupport {
+
+    @Autowired
+    private DaoHelper daoHelper;
+    private String lineageTreeNodeSequenceName;
+    private String insertLineageTreeNodeQuery;
+    private String deleteLineageTreeNodesQuery;
+    private String loadLineageTreeNodesQuery;
+    private String loadLineageTreeNodesByIdQuery;
+    private String loadLineageTreeNodesByReferenceIdQuery;
+
+    /**
+     * Persists new LineageTreeNode records.
+     * @param nodes {@code List<LineageTreeNode>} LineageTreeNodes to persist.
+     */
+    @Transactional(propagation = Propagation.MANDATORY)
+    public List<LineageTreeNode> save(final List<LineageTreeNode> nodes) {
+        if (!CollectionUtils.isEmpty(nodes)) {
+            List<Long> newIds = daoHelper.createIds(lineageTreeNodeSequenceName, nodes.size());
+            List<MapSqlParameterSource> params = new ArrayList<>(nodes.size());
```

Comment:
Can we use array here to avoid additional `params.toArray(new MapSqlParameterSource[nodes.size()])`?